src/lib.rs

```diff
@@ -79,6 +262,45 @@ impl Environment {
             self.vars
         }
     }
+
+    pub fn with_var<T: Into<Var>>(mut self, v: T) -> Environment {
+        let var = v.into();
+        self.vars.push((var.key, var.value));
+
+        self
+    }
+
+    pub fn var<T: ToString>(self, key: T) -> Var {
```

Comment:
1. Should `var(key).assign` replace `insert`?
2. Similarly to `with_var`, should this have a docstring?